src/test/kotlin/no/nav/pensjon/simulator/tpregisteret/TpregisteretClientTest.kt

```diff
@@ -0,0 +1,115 @@
+package no.nav.pensjon.simulator.tpregisteret
+
+import no.nav.pensjon.simulator.tech.trace.TraceAid
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.ArgumentMatchers.any
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mock
+import org.mockito.Mockito.`when`
+import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientRequestException
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+import java.net.URI
+
+@ExtendWith(MockitoExtension::class)
+class TpregisteretClientTest {
+
+    @Mock
+    private lateinit var webClientBuilder: WebClient.Builder
+
+    @Mock
+    private lateinit var webClient: WebClient
```

Comment:
Forslag: Bruk `MockWebServer` (slik det er gjort i [PenAlderspensjonClientTest](https://github.com/navikt/pensjonssimulator/blob/main/src/test/kotlin/no/nav/pensjon/simulator/alderspensjon/client/pen/PenAlderspensjonClientTest.kt)).
Fordelene er beskrevet i [Mocking a WebClient in Spring](https://www.baeldung.com/spring-mocking-webclient):

> We get the benefit of testing the intended HTTP interactions, and none of the challenges of mocking a complex fluent client.

Reply:
takk for forslaget og lenker 👍